compiler/src/iree/compiler/Bindings/Native/Transforms/test/wrap_entry_points.mlir

```diff
@@ -15,87 +15,87 @@
 // CHECK-SAME:     some.attr = 4 : index
-// CHECK: func.func private @_existingReflection
+// CHECK: util.func private @_existingReflection
 // CHECK-NOT: iree.reflection = {some.attr = 4 : index}
-func.func @existingReflection() attributes {
+util.func public @existingReflection() attributes {
   iree.reflection = {
     some.attr = 4 : index
   }
 } {
-  return
+  util.return
 }
 
 // -----
 
 // Tests that iree.abi.declaration is added when needed and otherwise the user
 // provided value is passed through.
 
-// CHECK-LABEL: func.func @existingDeclaration
+// CHECK-LABEL: util.func public @existingDeclaration
 // CHECK-SAME: iree.reflection =
 // CHECK-SAME:     iree.abi.declaration = "some.python.thing(types_are_overrated)"
-func.func @existingDeclaration(%arg0: tensor<i32>) attributes {
+util.func public @existingDeclaration(%arg0: tensor<i32>) attributes {
   iree.abi.declaration = "some.python.thing(types_are_overrated)"
 } {
-  return
+  util.return
 }
 
 // -----
 
 // Tests that name overrides propagate into both metadata and assertion IR.
 
-// CHECK-LABEL: func.func @namedEntry
+// CHECK-LABEL: util.func public @namedEntry
 // CHECK-SAME: iree.reflection =
 // CHECK-SAME:     iree.abi.declaration = "sync func @namedEntry(%my_input_0: tensor<3xf32>, %input1: tensor<3xf32>) -> (%my_output_0: tensor<3xf32>, %output1: tensor<3xf32>)"
-func.func @namedEntry(%arg0: tensor<3xf32> {iree.abi.name = "my_input_0"}, %arg1: tensor<3xf32>) ->
+util.func public @namedEntry(%arg0: tensor<3xf32> {iree.abi.name = "my_input_0"}, %arg1: tensor<3xf32>) ->
     (tensor<3xf32> {iree.abi.name = "my_output_0"}, tensor<3xf32>) {
   %0 = arith.addf %arg0, %arg1 : tensor<3xf32>
-  return %0, %0 : tensor<3xf32>, tensor<3xf32>
+  util.return %0, %0 : tensor<3xf32>, tensor<3xf32>
 }
 
 // -----
 
 // Tests that exports with encodings specified are propagated to the HAL ops.
 
-// CHECK-LABEL: func.func @exportEncodings
+// CHECK-LABEL: util.func public @exportEncodings
 //  CHECK-SAME:   iree.abi.declaration = "sync func @exportEncodings(%input0: tensor<?x8x8x3xf32> {iree.abi.encoding = tensor<?x8x8x3xi32>}) -> (%output0: tensor<?x8x8x3xf32> {iree.abi.encoding = tensor<?x8x8x3xi32>})"
 // CHECK: hal.tensor.import {{.+}} : !hal.buffer_view -> tensor<?x8x8x3xi32> as tensor<?x8x8x3xf32>{{.+}}
 // CHECK: hal.tensor.export {{.+}} : tensor<?x8x8x3xi32> as tensor<?x8x8x3xf32>{{.+}} -> !hal.buffer_view
 
-// CHECK-LABEL: func.func private @_exportEncodings
-func.func @exportEncodings(%arg0: tensor<?x8x8x3xf32> {iree.abi.encoding = tensor<?x8x8x3xi32>}) -> (tensor<?x8x8x3xf32> {iree.abi.encoding = tensor<?x8x8x3xi32>}) {
-  return %arg0 : tensor<?x8x8x3xf32>
+// CHECK-LABEL: util.func private @_exportEncodings
+util.func public @exportEncodings(%arg0: tensor<?x8x8x3xf32> {iree.abi.encoding = tensor<?x8x8x3xi32>}) -> (tensor<?x8x8x3xf32> {iree.abi.encoding = tensor<?x8x8x3xi32>}) {
+  util.return %arg0 : tensor<?x8x8x3xf32>
 }
```

Comment:
This sorts of tests will be even more IREE-specific now. A bit harder to hop between projects using MLIR. Doesn't seem like there is much we can do about that, given the clear benefits to using our own `func` and `return` ops though.

Reply:
the thing that wraps functions for IREE's ABI is IREE specific, and that's kind of the point